### PR TITLE
JA: translation xla/developing_new_backend

### PR DESCRIPTION
**Pull request by nuka137**

### CODE REVIEW

File: site/ja/xla/developing_new_backend.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+本ガイドは1つ1つ丁寧に説明したものではなく、LLVM、Bazel、TensorFlowの知識を前提としています。
+
+XLAは、新しいアーキテクチャやアクセラレータが、TensorFlowのグラフを処理するバックエンドを実装するための抽象的なインターフェースを提供します。
+XLAに対応することはとても簡単で、かつ新しいハードウェア向けに既存のTensorFlowのオペレーションを実装することと比較して拡張性もあります。
+
+実装のほとんどは、以下のシナリオのうちの1つに分類されます。
+
+1. LLVMのバックエンドが存在するかしないかにかかわらず、公式にXLAでサポートされていない既存のCPUアーキテクチャ
+2. LLVMのバックエンドが存在する、CPUではないハードウェア
+3. LLVMのバックエンドが存在する、CPUではないハードウェア
```
Comment (kiszk) [inline site/ja/xla/developing_new_backend.md]: [strong suggestion]
`存在する` ==> `存在しない`

File: site/ja/xla/developing_new_backend.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+## シナリオ2: LLVMのバックエンドが存在する、CPUではないハードウェア
+
+LLVM IRを出力する既存の [`xla::CPUCompiler`](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/compiler/xla/service/cpu/cpu_compiler.cc) や [`xla::GPUCompiler`](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/compiler/xla/service/gpu/nvptx_compiler.cc) クラスをベースとして、新しい [`xla::Compiler`](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/compiler/xla/service/compiler.h) を作ることが可能です。
+ハードウェアの性質によりLLVM IRの生成方法は異なりますが、多くのコードは既存のバックエンドと共有できるでしょう。
+
+よい参考例は、XLAの [GPUバックエンド](https://github.com/tensorflow/tensorflow/tree/master/tensorflow/compiler/xla/service/gpu) です。
+GPUのバックエンドはCPUとは異なるISAをターゲットとするため、GPUドメイン固有なコードの生成方法になります。
+
+
+## シナリオ3: LLVMのバックエンドが存在する、CPUではないハードウェア
```
Comment (kiszk) [inline site/ja/xla/developing_new_backend.md]: [strong suggestion]
`存在する` ==> `存在しない`

File: site/ja/xla/developing_new_backend.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+2. LLVMのバックエンドが存在する、CPUではないハードウェア
+3. LLVMのバックエンドが存在する、CPUではないハードウェア
+
+Note: LLVMのバックエンドとは、公式にリリースされたLLVMのバックエンド、または企業内で開発されたカスタマイズ版LLVMのバックエンドのことを指します。
+
+
+## シナリオ1: 公式にXLAでサポートされていない既存のCPUアーキテクチャ
+
+このシナリオの場合、既存の [XLA CPUバックエンド](https://github.com/tensorflow/tensorflow/tree/master/tensorflow/compiler/xla/service/cpu) を見ることから始めてください。
+XLAのCPUバックエンド間の主な違いは、LLVMによって生成されるコードであることから、XLAでは簡単にLLVMを使って異なるCPUをTensorFlowに対応できます。
```
Comment (kiszk) [inline site/ja/xla/developing_new_backend.md]: [weak suggestion]
`XLAでは簡単にLLVMを使って異なるCPUをTensorFlowに対応できます` ==> `XLAではLLVMを使って異なるCPUをTensorFlowに簡単に対応できます`

File: site/ja/xla/developing_new_backend.md [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+もしハードウェア企業がハードウェア向けのLLVMのバックエンドをもつ場合、ビルドされたLLVMのバックエンドをXLAに接続することは簡単です。
+JITモードでは、XLAのCPUバックエンドはホスト側のCPUのコードを生成します。
+Ahead-Of-Timeコンパイルでは、[`xla::AotCompilationOptions`](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/compiler/xla/service/compiler.h) が対象とするアーキテクチャに対して設定するLLVM Tripleを提供します。
+
+もし既存のLLVMのバックエンドがなくてもコード生成器が違う形で存在するならば、既存のCPUバックエンドの大部分を再利用できる可能性があります。
+
+
+## シナリオ2: LLVMのバックエンドが存在する、CPUではないハードウェア
+
+LLVM IRを出力する既存の [`xla::CPUCompiler`](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/compiler/xla/service/cpu/cpu_compiler.cc) や [`xla::GPUCompiler`](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/compiler/xla/service/gpu/nvptx_compiler.cc) クラスをベースとして、新しい [`xla::Compiler`](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/compiler/xla/service/compiler.h) を作ることが可能です。
```
Comment (kiszk) [inline site/ja/xla/developing_new_backend.md]: [weak suggestion]
`を作る` ==> `の実装を作る`

File: site/ja/xla/developing_new_backend.md [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+ハードウェアの性質によりLLVM IRの生成方法は異なりますが、多くのコードは既存のバックエンドと共有できるでしょう。
+
+よい参考例は、XLAの [GPUバックエンド](https://github.com/tensorflow/tensorflow/tree/master/tensorflow/compiler/xla/service/gpu) です。
+GPUのバックエンドはCPUとは異なるISAをターゲットとするため、GPUドメイン固有なコードの生成方法になります。
+
```
Comment (kiszk) [inline site/ja/xla/developing_new_backend.md]: Why is there no Japanese translation that corresponds to the following statement?
```
Other kinds of hardware, e.g. DSPs like Hexagon (which has an upstream LLVM backend), can reuse parts of the LLVM IR emission logic, but other parts will be unique.
```

File: site/ja/xla/developing_new_backend.md (hunk 1 above)
Comment (nuka137) [reply site/ja/xla/developing_new_backend.md]: Fixed it.

File: site/ja/xla/developing_new_backend.md (hunk 2 above)
Comment (nuka137) [reply site/ja/xla/developing_new_backend.md]: Fixed it.

File: site/ja/xla/developing_new_backend.md (hunk 3 above)
Comment (nuka137) [reply site/ja/xla/developing_new_backend.md]: Thanks for a suggestion. I fixed it.

File: site/ja/xla/developing_new_backend.md (hunk 4 above)
Comment (nuka137) [reply site/ja/xla/developing_new_backend.md]: Fixed it.

File: site/ja/xla/developing_new_backend.md (hunk 5 above)
Comment (nuka137) [reply site/ja/xla/developing_new_backend.md]: Thanks for the comment. I forgot translating this line.
Newest commit includes its translation.

**Comment by nuka137** [comment]
@kiszk 

Thanks for reviewing this translation.
I fixed all issues you mentioned.


@ohtaman @sfujiwara @masa-ita 

Could you review this PR?

File: site/ja/xla/developing_new_backend.md [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+この翻訳の品質を向上させるためのご意見をお持ちの方は、GitHubリポジトリ[tensorflow/docs](https://github.com/tensorflow/docs)にプルリクエストをお送りください。
+\
+コミュニティによる翻訳やレビューに参加していただける方は、
+[docs-ja@tensorflow.org メーリングリスト](https://groups.google.com/a/tensorflow.org/forum/#!forum/docs-ja)にご連絡ください。
+
+本予備ガイドは、効率的な方法でTensorFlowをハードウェアに容易に対応させたいと考えている、アーリーアダプターのためのものです。
+本ガイドは1つ1つ丁寧に説明したものではなく、LLVM、Bazel、TensorFlowの知識を前提としています。
+
+XLAは、新しいアーキテクチャやアクセラレータが、TensorFlowのグラフを処理するバックエンドを実装するための抽象的なインターフェースを提供します。
+XLAに対応することはとても簡単で、かつ新しいハードウェア向けに既存のTensorFlowのオペレーションを実装することと比較して拡張性もあります。
```
[the remark this answers is not in the document]
Comment (nuka137) [reply site/ja/xla/developing_new_backend.md]: Thanks for your suggestion.
I think your translation is better than mine.

File: site/ja/xla/developing_new_backend.md [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+
+
+## シナリオ2: LLVMのバックエンドが存在する、CPUではないハードウェア
+
+LLVM IRを出力する既存の [`xla::CPUCompiler`](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/compiler/xla/service/cpu/cpu_compiler.cc) や [`xla::GPUCompiler`](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/compiler/xla/service/gpu/nvptx_compiler.cc) クラスをベースとして、新しい [`xla::Compiler`](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/compiler/xla/service/compiler.h) の実装を作ることが可能です。
+ハードウェアの性質によりLLVM IRの生成方法は異なりますが、多くのコードは既存のバックエンドと共有できるでしょう。
+
+よい参考例は、XLAの [GPUバックエンド](https://github.com/tensorflow/tensorflow/tree/master/tensorflow/compiler/xla/service/gpu) です。
+GPUのバックエンドはCPUとは異なるISAをターゲットとするため、GPUドメイン固有なコードの生成方法になります。
+他の種類のハードウェア、例えば（アップストリームのLLVMのバックエンドを持つ）HexagonのようなDSPは、LLVM IRの生成論理部を再利用することができますが、他の部分は固有のものになるでしょう
```
[the remark this answers is not in the document]
Comment (nuka137) [reply site/ja/xla/developing_new_backend.md]: Your translation is more natural than mine. I fixed it.

File: site/ja/xla/developing_new_backend.md [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+
+LLVMを利用できない場合、対象のハードウェア向けに新しいバックエンドを実装することが最良の選択肢となります。
+この選択肢は、多大な労力を必要とします。
+実装しなければならないクラスは次に示すとおりです。
+
+* [StreamExecutor](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/stream_executor/stream_executor.h): 多くのデバイスでは、`StreamExecutor` のすべてのメソッドが必要になることはありません。詳細は既存の `StreamExecutor` の実装を見てください。
+* [xla::Compiler](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/compiler/xla/service/compiler.h): 本クラスは、HLO Computationから `xla::Executable` へのコンパイル処理を隠蔽します。
+* [xla::Executable](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/compiler/xla/service/executable.h): 本クラスは、コンパイル済みのComputationをプラットフォーム上で実行するために使用されます。
+* [xla::TransferManager](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/compiler/xla/service/transfer_manager.h): 本クラスは、与えられたデバイスメモリのハンドルからXLAのリテラルデータを構築するための、プラットフォーム特有のしくみを提供することを可能にします。言い換えれば、ホストからデバイスまたはその反対のデータ転送処理を隠蔽します。
+
```
[the remark this answers is not in the document]
Comment (nuka137) [reply site/ja/xla/developing_new_backend.md]: I fixed it.

**Comment by nuka137** [comment]
@sfujiwara 

Thanks for reviewing.
I fixed all issues you mentioned.

**Comment by sfujiwara** [comment]
@nuka137 Thanks, great!!

@lamberta I approved this PR. Please merge it!

**Pull request merged**